src/ucs/datastruct/list.h

```diff
@@ -98,6 +98,28 @@ static inline void ucs_list_insert_before(ucs_list_link_t *pos,
     ucs_list_insert_replace(pos->prev, pos, new_link);
 }
 
+/**
+ * Insert an item to a list at a specific index.
+ *
+ * @param head        List head.
+ * @param new_link    Item to insert.
+ * @param index       Index at which to insert.
+ */
+static inline void ucs_list_insert_at_index(ucs_list_link_t *head,
```

Comment:
I think it would make sense to add a few unit tests for this function covering corner cases: insert first, last, empty list etc